### PR TITLE
Issue #2585429: Replacing evaluate() with doEvaluate() in NodeIsStick…

### PR DESCRIPTION
**Pull request by emclaughlin1215**
…y condition.

See d.o issues: https://www.drupal.org/node/2585429 and https://www.drupal.org/node/2572119

### CODE REVIEW

File: src/Plugin/Condition/NodeIsSticky.php [hunk 1]
```diff
@@ -27,10 +28,15 @@
...
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node to check for stickiness.
+   *
+   * @return bool
+   *   TRUE if the node is sticky.
    */
-  public function evaluate() {
-    $node = $this->getContextValue('node');
+  public function doEvaluate(NodeInterface $node) {
```
Comment (klausi) [inline src/Plugin/Condition/NodeIsSticky.php]: should be protected, not public.

**Comment by emclaughlin1215** [comment]
Addressed the comments and pushed up changes.

File: src/Plugin/Condition/NodeIsSticky.php [hunk 2]
```diff
@@ -27,10 +28,15 @@
  */
 class NodeIsSticky extends RulesConditionBase {
   /**
-   * {@inheritdoc}
+   * Check if the given node is sticky.
+   *
+   * @param \Drupal\node\NodeInterface $node
+   *   The node to check for stickiness.
```
Comment (klausi) [inline src/Plugin/Condition/NodeIsSticky.php]: "The node to check." should be enough.

**Comment by emclaughlin1215** [comment]
Updated based on comments.

**Comment by klausi** [comment]
merged, thanks!

**Pull request closed without merging**